litestar/_signature/utils.py

```diff
@@ -56,6 +57,7 @@ def create_signature_model(
         fn: A callable.
         preferred_validation_backend: Validation/Parsing backend to prefer, if installed
         parsed_signature: A parsed signature for the handler/dependency function.
+        has_data_dto: Is a data DTO defined for the handler?
```

Comment:
Added this param so that when a dto is available for the handler the "data" field on the signature model will be typed `Any`. I think this is correct behavior as the DTO is doing the raw data validation, so we just want whatever it returns to be passed through to the handler.

I noticed this when testing the behavior of the DTOs with `underscore_fields_private=False`, the signature model would emit

>   /home/peter/PycharmProjects/litestar/litestar/_signature/models/pydantic_signature_model.py:165: RuntimeWarning: fields may not start with an underscore, ignoring "_baz"
    model: type[PydanticSignatureModel] = create_model(  # type: ignore

Reply:
Have you tested the OpenAPI schema?

Reply:
Yes, schema for the request body is requested from the DTO if one has been declared:

https://github.com/litestar-org/litestar/blob/5d6cc88a06ff0e495467bde7468a5a60fd551003/litestar/_openapi/request_body.py#L33-L34

From this dataclass definition and the two handlers, one with and one without a DTO applied:

```py

@dataclass
class Thing:
    a: int
    _b: int


@post("/without")
async def without_dto(data: Thing) -> Thing:
    """Route Handler that outputs hello world."""
    return data


@post("/with", dto=DataclassDTO[Thing])
async def with_dto(data: Thing) -> Thing:
    """Route Handler that outputs hello world."""
    return data

```

Without docs:

![image](https://github.com/litestar-org/litestar/assets/20659309/b76171cb-e5c8-47e0-b0d2-d2d263dd6855)

With docs:

![image](https://github.com/litestar-org/litestar/assets/20659309/a7945daf-4043-44b6-af04-e684f32d861c)